R/derive_vars_joined_summary.R

```diff
@@ -0,0 +1,494 @@
+#'   or `first_cond_upper` is specified or `join_type` equals `"before"` or
+#'   `"after"`.
+#'
+#' @permitted [msg_type]
+#'
+#' @inheritParams get_joined_data
+#' @inheritParams derive_vars_merged
+#'
+#' @details
+#'
+#' 1. The variables specified by `order` are added to the additional dataset
+#' (`dataset_add`).
+#'
+#' 1. The variables specified by `join_vars` are added to the additional dataset
+#' (`dataset_add`).
+#'
+#' 1. The records from the additional dataset (`dataset_add`) are restricted to
+#' those matching the `filter_add` condition.
+#'
+#' 1. The input dataset and the (restricted) additional dataset are left joined
+#' by the grouping variables (`by_vars`). If no grouping variables are
+#' specified, a full join is performed.
+#'
+#' 1. If `first_cond_lower` is specified, for each observation of the input
+#'     dataset the joined dataset is restricted to observations from the first
+#'     observation where `first_cond_lower` is fulfilled (the observation fulfilling
+#'     the condition is included) up to the observation of the input dataset. If for
+#'     an observation of the input dataset the condition is not fulfilled, the
+#'     observation is removed.
+#'
+#'     If `first_cond_upper` is specified, for each observation of the input
+#'     dataset the joined dataset is restricted to observations up to the first
+#'     observation where `first_cond_upper` is fulfilled (the observation
+#'     fulfilling the condition is included). If for an observation of the input
+#'     dataset the condition is not fulfilled, the observation is removed.
+#'
+#'     For an example see the last example in the "Examples" section.
+#'
+#' 1. The joined dataset is restricted by the `filter_join` condition.
+#'
+#' 1. The variables specified for `new_vars` are created and merged to the input
+#' dataset. I.e., the output dataset contains all observations from the input
+#' dataset. For observations without a matching observation in the joined
+#' dataset the new variables are set as specified by `missing_values` (or to
+#' `NA` for variables not in `missing_values`). Observations in the additional
+#' dataset which have no matching observation in the input dataset are ignored.
+#'
+#' `r roxygen_save_memory()`
+#'
+#' @return The output dataset contains all observations and variables of the
+#'   input dataset and additionally the variables specified for `new_vars`
+#'   derived from the additional dataset (`dataset_add`).
+#'
+#' @seealso [derive_vars_joined()], [derive_var_merged_summary()],
+#' [derive_var_joined_exist_flag()], [filter_joined()]
+#'
+#' @keywords der_gen
+#' @family der_gen
+#'
+#' @export
+#'
+#' @examplesx
+#'
+#' @info The examples focus on the functionality specific to this function. For
+#'   examples of functionality common to all "joined" functions like
+#'   `filter_join`, `join_type`, `first_cond_lower`, ... please see the examples
```

Comment:
This is not accurate anymore as you have (simple) examples for `join_type` and `first_cond_*`

Reply:
Good point! Updated.